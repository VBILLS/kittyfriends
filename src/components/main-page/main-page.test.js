import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainPage from './main-page.component';

describe('MainPage tests', () => {
  let wrapper;
  beforeEach(() => {
    const mockProps = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: '',
      isPending: false
    };
    wrapper = shallow(<MainPage {...mockProps} />);
  });

  it('renders mainpage without crashing', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('filters robots correctly', () => {
    const mockProps2 = {
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: 'John',
          email: 'test@gmail.com'
        }
      ],
      searchField: 'john',
      isPending: false
    };
    const newWrapper = shallow(<MainPage {...mockProps2} />);
    expect(wrapper.instance().filterRobots()).toEqual([]);
    expect(newWrapper.instance().filterRobots()).toEqual([
      {
        id: 3,
        name: 'John',
        email: 'test@gmail.com'
      }
    ]);
  });

  it('has isPending True with Loading...', () => {
    const mockProps3 = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: '',
      isPending: true
    };
    const wrapper3 = shallow(<MainPage {...mockProps3} />);
    expect(wrapper3.exists('.loading')).toBe(true);
  });
});
