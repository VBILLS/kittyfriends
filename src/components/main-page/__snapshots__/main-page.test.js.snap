// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage tests renders mainpage without crashing 1`] = `
<div>
  <Header />
  <CounterButton />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
